### PR TITLE
chore: Stripped dead code

### PR DESCRIPTION
**Pull request by semty**
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [ ] This implements/fixes issue #.

---

**Description:**

Stripped dead code throughout the project. 
It's a good practice to perform from time to time, leaving only the code that is actually in use. 

**Note:**

I'm not sure whether this project still accepts pull requests from not core team members. 
@low-batt, it seems to me that you're the most active participant lately, could you, please, help me with that?

### CODE REVIEW

**Comment by low-batt** [comment]
@Semty, you definitely do not need to be one of IINA's core developers to propose changes.

IINA is slow at reviewing and merging PRs as the senior developers are very busy. Usually it occurs in spurts when those developers have time to focus on IINA.

I am a newbie junior contributor. I don't merge, but I do review PRs. No promises, but I will try and find time to take a look at this one this weekend. This really needs a review by the senior developers as it covers areas of IINA I am unfamiliar with.

At the moment the senior developers are focused on getting a release out. The release branch as already been cut. The hold up is the need to patch some problems in the updated dependencies.

Commits going into `develop` now will be going into the following release. Which is why this is sort of a good time to clear out dead code. One concern is that there is resistance to refactoring/cleanup with so many PRs outstanding over the fear of introducing lots of merge conflicts.

While on the topic of contributing I should point out that although [CONTRIBUTING](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md) indicates the latest version of Xcode must be used, at the moment we are building with Xcode 13 as well as Xcode 14 in order to support macOS 10.11 and 10.12, which Xcode 14 dropped.

**Comment by semty** [comment]
> @Semty, you definitely do not need to be one of IINA's core developers to propose changes.
> 
> 
> 
> IINA is slow at reviewing and merging PRs as the senior developers are very busy. Usually it occurs in spurts when those developers have time to focus on IINA.
> 
> 
> 
> I am a newbie junior contributor. I don't merge, but I do review PRs. No promises, but I will try and find time to take a look at this one this weekend. This really needs a review by the senior developers as it covers areas of IINA I am unfamiliar with.
> 
> 
> 
> At the moment the senior developers are focused on getting a release out. The release branch as already been cut. The hold up is the need to patch some problems in the updated dependencies.
> 
> 
> 
> Commits going into `develop` now will be going into the following release. Which is why this is sort of a good time to clear out dead code. One concern is that there is resistance to refactoring/cleanup with so many PRs outstanding over the fear of introducing lots of merge conflicts.
> 
> 
> 
> While on the topic of contributing I should point out that although [CONTRIBUTING](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md) indicates the latest version of Xcode must be used, at the moment we are building with Xcode 13 as well as Xcode 14 in order to support macOS 10.11 and 10.12, which Xcode 14 dropped.

@low-batt hey! Got it, thank you so much for your thorough response 👌. 
Let me know if you find anything that should not be removed, I'm open to fix it if needed

File: iina/MPVCommand.swift
Comment (low-batt) [inline iina/MPVCommand.swift]: My rule is that generated files should never be checked into source control. They should always be regenerated as a part of the build. Too often I have encountered generated files where developers modified the generated files instead of correcting the true source or the generator that created the generated source. I have even countered the situation where the true source and the generator have been lost.

IINA is violating this rule. The swift classes `MPVCommand`, `MPVOption` and `MPVProperty` are all generated and must not be modified by developers. These files are generated from mpv documentation by the script `other/parse_doc.rb` and then checked into git. This is done whenever IINA upgrades to a newer version of the mpv library.

Another of my own rules is being violated, which is that all source code generator must start the generated code with a BIG comment that tries to make it clear to developers that this is a generated file that must not be directly modified by developers. Possibly that comment should contain details about the generation. In this case it probably should indicate the mpv version the source was generated from.

Within Xcode these sources aren't even in their own "Generated" group. The [Current structure](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md#current-structure) section of the [Contributing](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md) document does not warn contributors about these generated sources.

I just entered issue #4429 that suggests improvements in this area.

File: iina/MPVController.swift
Comment (low-batt) [inline iina/MPVController.swift:1363]: Will have to see what the senior developers think, but I'm strongly against removing the name of parameters, even if they are not used. After this change a developer reading this code can't tell what it is that is being passed and ignored.

File: iina/MPVOption.swift
Comment (low-batt) [inline iina/MPVOption.swift]: Generated source, must not be modified.

File: iina/MPVProperty.swift
Comment (low-batt) [inline iina/MPVProperty.swift]: Generated source, must not be modified.

File: iina/ShooterSubtitle.swift
Comment (low-batt) [inline iina/ShooterSubtitle.swift]: This was added in anticipation of log messages being added by someone who knows this provider.

There are problems with not being able to tell what is going wrong when trying to find and download subtitles. Some refactoring was done to share logging and reporting in `OnlineSubtitle`. However as can be seen in `OpenSubSubtitle` some logging should be done by the specific provider class.

File: iina/VideoView.swift
Comment (low-batt) [inline iina/VideoView.swift:459]: Although unused with respects to the code, these parameter names effectively provide documentation for developers reading the code. See what the senior developers say, but I am against removing parameter names.

File: iina/ViewLayer.swift
Comment (low-batt) [inline iina/ViewLayer.swift:199]: As the comments indicates this is a debugging tool for developers and must not be removed. I'm thinking the correct change may be to use a Swift [Conditional Compilation Block](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Conditional-Compilation-Block) to avoid including dead code. See what the senior developers think.

**Review by low-batt — CHANGES_REQUESTED** [review]
I pulled the PR locally, built IINA and confirmed IINA started and was able to play a video.

I'm unfamiliar with a lot of this code, so the real review must be done by the senior developers. I had planned on not requesting changes and leaving it to those developers to request changes as I am unsure as to whether they would agree with some of my comments. However…

I'm certain this PR in its current state will be rejected due to including changes to generated sources. This is an existing pitfall for new contributors. I have entered issue #4429 for the failure to properly identify generated sources. See that issue and my inline comment.

On my other comments, feel free to wait and see what the other developers think.

Thank you for proposing these changes to IINA!

File: iina/MPVCommand.swift
Comment (semty) [reply iina/MPVCommand.swift]: @low-batt hey, thanks for your review! I didn't really know that these classes are auto-generated, I agree with your take here. 
Fixed in ce4a8bca1c16f0f7077a75b0fbe75b010cc44c95

File: iina/MPVOption.swift
Comment (semty) [reply iina/MPVOption.swift]: Fixed in ce4a8bca1c16f0f7077a75b0fbe75b010cc44c95

File: iina/MPVProperty.swift
Comment (semty) [reply iina/MPVProperty.swift]: Fixed in ce4a8bca1c16f0f7077a75b0fbe75b010cc44c95

File: iina/VideoView.swift
Comment (semty) [reply iina/VideoView.swift:459]: Hmm, I tend to disagree here. My two points:
1. In my opinion, it actually better to omit some stuff that you don't need to perform your logic, you have less things to grasp and see only the important ones
2. If you want to change/debug this piece of code, it's better if you're familiar with it a bit more than just knowing the parameter names. It can be done really easily by checking where this function is used (`CVDisplayLinkSetOutputCallback`) and locating the parameter (`CVDisplayLinkOutputCallback`). After that you'll see the complete documentation with everything you may need

But, of course, I understand that it can be a hot topic, so I'm fully okay to return parameter names in every function even if it's not used

File: iina/ViewLayer.swift
Comment (semty) [reply iina/ViewLayer.swift:199]: @low-batt for the time being, I just returned the old code, fixed here: 
- c6e3bbc32ef70daf08139d84cfb6828ce61aebaa

File: iina/ShooterSubtitle.swift
Comment (semty) [reply iina/ShooterSubtitle.swift]: @low-batt Fixed: 6b202ba4703d6a8885379596bb79c0a96e4fa2e2

File: iina/VideoView.swift
Comment (low-batt) [reply iina/VideoView.swift:459]: On this one lets see what the senior developers think.

File: iina/ViewLayer.swift
Comment (low-batt) [reply iina/ViewLayer.swift:199]: Sounds good.

File: iina/AppDelegate.swift
Comment (saagarjha) [inline iina/AppDelegate.swift:18]: As far as I can tell this was never used, perhaps worth adjusting the code to reflect that?

File: iina/Aspect.swift
Comment (saagarjha) [inline iina/Aspect.swift]: We should probably keep this even if it's unused

File: iina/CollapseView.swift
Comment (saagarjha) [inline iina/CollapseView.swift:58]: You can't remove this parameter, see https://developer.apple.com/library/archive/documentation/General/Conceptual/Devpedia-CocoaApp/TargetAction.html#//apple_ref/doc/uid/TP40009071-CH3-SW2

File: iina/Extensions.swift
Comment (saagarjha) [inline iina/Extensions.swift:158]: Probably worth keeping this but with the right parameter name

File: iina/InfoDictionary.swift
Comment (saagarjha) [inline iina/InfoDictionary.swift:79]: Pretty sure this was just added recently, @low-batt you should probably look at this and see if you needed it

File: iina/JavascriptPlugin.swift
Comment (saagarjha) [inline iina/JavascriptPlugin.swift:340]: cc @lhc70000 did you intend to do something with this

**Review by saagarjha — CHANGES_REQUESTED** [review]
I appreciate konmari efforts. Lots of this needs to go. Deciding which parts, though, is complicated. In general, these things should probably not be removed:

* Sets of related properties or methods that exist for completeness: e.g. if we have setString, setInt, setDouble, … it doesn't make sense to remove one of these because it's not used. People expect the full set of APIs to be available.
* Things that were just added and intend to be used in the near future. Determining this will probably be hard. cc @lhc70000 @uiryuu since they should sign off on this.

Marking for changes because some of these definitely need to be rolled back–see the comments for the ones I spotted.

File: iina/CollapseView.swift
Comment (semty) [reply iina/CollapseView.swift:58]: @saagarjha hey! thank you for your review! 
> You can't remove this parameter, see https://developer.apple.com/library/archive/documentation/General/Conceptual/Devpedia-CocoaApp/TargetAction.html#//apple_ref/doc/uid/TP40009071-CH3-SW2

are you sure? it works on iOS to omit the sender

File: iina/Aspect.swift
Comment (semty) [reply iina/Aspect.swift]: @saagarjha fixed: https://github.com/iina/iina/pull/4425/commits/1d610bdd56709994411939c29b895d90177539c1

File: iina/Extensions.swift
Comment (semty) [reply iina/Extensions.swift:158]: @saagarjha fixed: https://github.com/iina/iina/pull/4425/commits/676b2847047bc273c39582aa00f1308e45ef1373

File: iina/InfoDictionary.swift
Comment (low-batt) [reply iina/InfoDictionary.swift:79]: Yes, this was recently added as a part of issue #4383 / PR #4385. However my role in that issue was to just get it started and hand it off to @uiryuu. Will have to ask @uiryuu about this change.

File: iina/CollapseView.swift
Comment (saagarjha) [reply iina/CollapseView.swift:58]: It's [legal on iOS](https://developer.apple.com/documentation/uikit/uicontrol#1658488), but not on macOS. Things will mostly still work because the argument register will probably not get touched but it's good to do it correctly.

File: iina/Extensions.swift
Comment (uiryuu) [inline iina/Extensions.swift:133]: Remove spaces

File: iina/InfoDictionary.swift
Comment (uiryuu) [reply iina/InfoDictionary.swift:79]: Let keep this, although there isn't any code using this. We might want to use this in the future.

File: iina/MPVController.swift
Comment (uiryuu) [reply iina/MPVController.swift:1363]: +1 for not changing

File: iina/ViewLayer.swift
Comment (uiryuu) [inline iina/ViewLayer.swift:198]: Space

**Review by uiryuu — CHANGES_REQUESTED** [review]

File: iina/MPVController.swift
Comment (uiryuu) [inline iina/MPVController.swift:571]: These functions were added intentionally, in the future we might want to use them. Please convert this back.

File: iina/Aspect.swift
Comment (uiryuu) [inline iina/Aspect.swift:14]: Space

File: iina/Aspect.swift
Comment (uiryuu) [inline iina/Aspect.swift:23]: Space

File: iina/VideoView.swift
Comment (uiryuu) [reply iina/VideoView.swift:459]: I'd like to keep it as it is.

**Review by uiryuu — CHANGES_REQUESTED** [review]
Just drop some comments here, will take a deeper look another day. Merge conflicts look scary but are surprisingly trivial.

**Comment by uiryuu** [comment]
Merged via https://github.com/iina/iina/commit/f121558adb89d8b9ac55e313bbb65cf71898d2d9. Thanks @Semty!

**Pull request closed without merging**
